src/constructs/cloudwatch/ec2-alarms.ts

```diff
@@ -0,0 +1,50 @@
+import { ComparisonOperator, MathExpression, TreatMissingData } from "@aws-cdk/aws-cloudwatch";
+import { HttpCodeElb, HttpCodeTarget } from "@aws-cdk/aws-elasticloadbalancingv2";
+import type { GuStack } from "../core";
+import type { AppIdentity } from "../core/identity";
+import type { GuApplicationLoadBalancer } from "../loadbalancing";
+import type { GuAlarmProps } from "./alarm";
+import { GuAlarm } from "./alarm";
+
+/**
+ * Creates an alarm which is triggered whenever the percentage of requests with a 5xx response code exceeds
+ * the specified threshold.
+ */
+export interface Gu5xxPercentageMonitoringProps
+  extends Omit<GuAlarmProps, "evaluationPeriods" | "metric" | "period" | "threshold" | "treatMissingData">,
+    AppIdentity {
+  tolerated5xxPercentage: number;
+  numberOfFiveMinutePeriodsToEvaluate?: number;
```

Comment:
AWS default to 5-minute evaluation periods - we could consider overriding this default (or allowing users to) if we felt that it was important?

Reply:
I think 5 minutes is too slow for PROD. Would favour GuCDK providing a higher frequency, but allow it to be overridden.

IIUC the alarm will trip if `tolerated5xxPercentage` is exceeded within `numberOfFiveMinutePeriodsToEvaluate`? Might be overkill, but maybe we can print a warning when this is likely to be too long to be useful and other alarming strategies should be considered.

Reply:
> I think 5 minutes is too slow for PROD. Would favour GuCDK providing a higher frequency, but allow it to be overridden.

I'm inclined to agree - perhaps 1 minute would be better? 

We are using the [same default for lambda-based alerts](https://github.com/guardian/cdk/blob/1c648dcd8c0a0ed42096080c69e7405571031e21/src/constructs/cloudwatch/lambda-alarms.ts#L13). I will have a look at improving granularity for both in a subsequent PR, as it will require a bit of further experimentation/testing.